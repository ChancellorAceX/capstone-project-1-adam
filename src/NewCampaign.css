* {
  box-sizing: border-box;
}

.inputSections {
  width:70vw;
  display:flex;
  justify-content: space-around;
  align-items: center;
}

.locations label {
  margin:0;
}

.tableBox {
  width:70vw;
  display:flex;
  justify-content: space-around;
  margin-top:2.5vh;
}

.tableBox th,td {
  border:1px solid black;
  padding:5px;
}

.tableBox ul {
  margin:0;
  padding-left:20px;
}