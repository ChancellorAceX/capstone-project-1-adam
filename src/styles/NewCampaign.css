main.newCampaign {
  width:calc(100vw + 15px);
}

form.newCampaign {
  position:fixed;
  top:10vh;
  padding:10px;
  height:82vh;
  background-color: black;
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow:auto;
}

h3.characterInputs {
  color:#b40a0a;
  text-align:center;
}

section.inputSections {
  display:flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
}

button.addPlayer {
  font-weight: 600;
  width:40vw;
  border-radius:20px;
  background-color: #b40a0a;
  color: #aaaaaa;
  text-shadow:-1px -1px 0 #444444, 1px 1px 0 #444444,1px -1px 0 #444444, -1px 1px 0 #444444; 
}

section.inputSections * {
  margin:0;
  margin-bottom: 10px;
  width:70vw;
}

section.inputSections input {
  width:60vw;
}

main.newCampaign table {
  width:90vw;
  color:#aaaaaa;
}

section.tableBox {
  margin-top:15px;
  width:95vw;
  display:flex;
  flex-direction: column;
  align-items: center;
}

section.Campaign {
  width:100vw;
  position:fixed;
  bottom: 8vh;
  border-top: 1px solid #aaaaaa;
  padding-top:4px;
  padding-bottom:4px;
  display:flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  background-color: black;
}

section.Campaign button {
  font-weight: 600;
  width:40vw;
  border-radius:20px;
  background-color: #b40a0a;
  color: #aaaaaa;
  text-shadow:-1px -1px 0 #444444, 1px 1px 0 #444444,1px -1px 0 #444444, -1px 1px 0 #444444;
}