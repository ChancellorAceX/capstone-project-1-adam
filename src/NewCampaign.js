import React from 'react';
import './NewCampaign.css';

export default function NewCampaign(){
  return(
    <section className='container'>
      <h2>New Campaign</h2>
      <form className='newCampaign'>
        <input className='title' placeholder='Campaign Title'/>
        <section className='inputSections'>
          <section className='locations'>
            <input className='locationTitle' placeholder='Location Name'/>
            <label for='locationList'>Nest under:</label>
            <select id='locationList' className='Sub-location'>
              <option>None: Main Location</option>
              <option>Location 1</option>
              <option>Location 2</option>
              <option>Location 3</option>
            </select>
            <button className='addLocation'>+ Location</button>
          </section>
          <section className='players'>
            <input className='playerName' placeholder='Name'/>
            <input classNAme='playerClass' placeholder='Class'/>
            <input className='playerLevel' placeholder='Level'/>
            <button className='addPlayer'>+ Player</button>
          </section>
        </section>
        <button type='submit'>+ Save Campaign</button>
      </form>
      <section className='tableBox'>
        <table className='locations left'>
          <tr>
            <th>
              <h4>Location 1</h4>
            </th>
            <td>
              <ul>
                <li>Sub-location 1</li>
                <li>Sub-location 2</li>
                <li>Sub-location 3</li>
              </ul>
            </td>
          </tr>
          <tr>
            <th>
              <h4>Location 2</h4>
            </th>
            <td>
              <ul>
                <li>Sub-location 4</li>
                <li>Sub-location 5</li>
                <li>Sub-location 6</li>
              </ul>
            </td>
          </tr>
          <tr>
            <th>
              <h4>Location 3</h4>
            </th>
            <th>
              <ul>

              </ul>
            </th>
          </tr>
        </table>
        <table className='players right'>
          <tr>
            <th>Character</th>
            <th>Class: Level</th>
          </tr>
          <tr>
            <td>Nire</td>
            <td>Druid: 4</td>
          </tr>
          <tr>
            <td>Bert</td>
            <td>Wizard: 4</td>
          </tr>
          <tr>
            <td>Pizzacart</td>
            <td>Paladin: 3</td>
          </tr>
          <tr>
            <td>Xanaphia</td>
            <td>Warlock: 3</td>
          </tr>
        </table>
      </section>
    </section>
  )
}