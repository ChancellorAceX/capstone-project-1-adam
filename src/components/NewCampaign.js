import React from 'react';
import { withRouter } from 'react-router-dom';
import Header from './Header';
import NewCampaignCharacter from './NewCampaignCharacter';
import '../styles/NewCampaign.css';

export default withRouter(function NewCampaign(props) {
  return (
    <section className='container'>
      <Header
        pageName='New Campaign'
        history={props.history}
        loggedInToggle={props.loggedInToggle}
      />
      <main className='newCampaign'>
        <form className='newCampaign' onSubmit={props.saveCampaign}>
          <section className='inputSections'>
            <h3 className='characterInputs'>Add Characters</h3>
            <input className='character' type='text' onChange={props.setAddCharacter} placeholder='Name' value={props.addCharacter.character} />
            <input className='pcclass' placeholder='Class' type='text' onChange={props.setAddCharacter} value={props.addCharacter.pcclass} />
            <input className='level' placeholder='Level' type='number' onChange={props.setAddCharacter} value={props.addCharacter.level} />
            <input className='ac' id='ac' placeholder='AC' type='number' onChange={props.setAddCharacter} value={props.addCharacter.ac} />
            <input className='maxhp' placeholder='Max HP' type='number' onChange={props.setAddCharacter} value={props.addCharacter.maxhp} />
            <button className='addPlayer' onClick={props.addToAddCharacters}>+ Player</button>
          </section>
          <section className='tableBox'>
            <table className='players'>
              <tbody>
                {props.addCharacters.map((character, i) =>
                  <NewCampaignCharacter
                    key={`character${i}`}
                    character={character}
                  />
                )}
              </tbody>
            </table>
          </section>
          <section className='Campaign'>
            <input className='title' placeholder='Campaign Title' onChange={props.setAddCampaign} value={props.addCampaign} required />
            <button type='submit'>+ Save Campaign</button>
          </section>
        </form>

      </main>
    </section>
  );
});