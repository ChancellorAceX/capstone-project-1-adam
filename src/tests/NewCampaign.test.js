import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewCampaign from '../components/NewCampaign';

describe('NewCampaign component', () => {
const props = {
  "addCampaign": "",
  "setAddCampaign": "ƒ () {}",
  "addCharacters": [
    "{ac: \"AC\", character: \"Character\", level: \"Level\", …}"
  ],
  "addToAddCharacters": "ƒ () {}",
  "addCharacter": {
    "character": "",
    "pcclass": "",
    "ac": "",
    "level": "",
    "maxhp": ""
  },
  "setAddCharacter": "ƒ () {}",
  "saveCampaign": "ƒ () {}",
  "history": {
    "length": 20,
    "action": "PUSH",
    "location": "{hash: \"\", key: \"m9myf9\", pathname: \"/new/campaign\"…}",
    "createHref": "ƒ createHref() {}",
    "push": "ƒ push() {}",
    "replace": "ƒ replace() {}",
    "go": "ƒ go() {}",
    "goBack": "ƒ goBack() {}",
    "goForward": "ƒ goForward() {}",
    "block": "ƒ block() {}",
    "listen": "ƒ listen() {}"
  },
  "loggedInToggle": "ƒ () {}",
  "location": {
    "pathname": "/new/campaign",
    "search": "",
    "hash": "",
    "key": "m9myf9"
  },
  "match": {
    "path": "/new/campaign",
    "url": "/new/campaign",
    "isExact": true,
    "params": "{}"
  }
};

  it('renders the full new campaign form component', () => {
    const wrapper = shallow(<NewCampaign {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});