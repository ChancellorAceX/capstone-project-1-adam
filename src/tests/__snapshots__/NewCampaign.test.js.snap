// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCampaign component renders the full new campaign form component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
